packages/cli/src/PublicApi/v1/handlers/workflows/workflows.service.ts

```diff
@@ -0,0 +1,143 @@
+
+export async function getSharedWorkflowIds(user: User): Promise<number[]> {
+	const sharedWorkflows = await Db.collections.SharedWorkflow.find({
+		where: {
+			user,
+		},
+	});
+	return sharedWorkflows.map((workflow) => workflow.workflowId);
+}
+
+export async function getSharedWorkflow(
+	user: User,
+	workflowId?: string | undefined,
+): Promise<SharedWorkflow | undefined> {
+	const sharedWorkflow = await Db.collections.SharedWorkflow.findOne({
+		where: {
+			...(!isInstanceOwner(user) && { user }),
+			...(workflowId && { workflow: { id: workflowId } }),
+		},
+		relations: ['workflow'],
+	});
+	return sharedWorkflow;
+}
+
+export async function getSharedWorkflows(
+	user: User,
+	options: {
+		relations?: string[];
+		workflowIds?: number[];
+	},
+): Promise<SharedWorkflow[]> {
+	const sharedWorkflows = await Db.collections.SharedWorkflow.find({
+		where: {
+			...(!isInstanceOwner(user) && { user }),
+			...(options.workflowIds && { workflow: { id: In(options.workflowIds) } }),
+		},
+		...(options.relations && { relations: options.relations }),
+	});
+	return sharedWorkflows;
+}
+
+export async function getWorkflowById(id: number): Promise<WorkflowEntity | undefined> {
+	const workflow = await Db.collections.Workflow.findOne({
+		where: {
+			id,
+		},
+	});
+	return workflow;
+}
+
+export async function getWorkflowIdsWithTags(tags: string[]): Promise<number[]> {
+	const dbTags = await Db.collections.Tag.find({
+		where: {
+			name: In(tags),
+		},
+		relations: ['workflows'],
+	});
+
+	const workflowIdsPerTag = dbTags.map((tag) => tag.workflows.map((workflow) => workflow.id));
+
+	return intersection(...workflowIdsPerTag);
+}
+
+export async function createWorkflow(
+	workflow: WorkflowEntity,
+	user: User,
+	role: Role,
+): Promise<WorkflowEntity> {
+	let savedWorkflow: unknown;
+	await Db.transaction(async (transactionManager) => {
+		const newWorkflow = new WorkflowEntity();
+		Object.assign(newWorkflow, workflow);
+		savedWorkflow = await transactionManager.save<WorkflowEntity>(newWorkflow);
+		const newSharedWorkflow = new SharedWorkflow();
+		Object.assign(newSharedWorkflow, {
+			role,
+			user,
+			workflow: savedWorkflow,
+		});
+		await transactionManager.save<SharedWorkflow>(newSharedWorkflow);
+	});
+	return savedWorkflow as WorkflowEntity;
+}
+
+export async function activeWorkflow(workflow: WorkflowEntity): Promise<void> {
+	await Db.collections.Workflow.update(workflow.id, { active: true, updatedAt: new Date() });
+}
+
+export async function desactiveWorkflow(workflow: WorkflowEntity): Promise<void> {
+	await Db.collections.Workflow.update(workflow.id, { active: false, updatedAt: new Date() });
+}
+
+export async function deleteWorkflow(workflow: WorkflowEntity): Promise<void> {
+	await Db.collections.Workflow.remove(workflow);
+}
+
+export async function getWorkflows(
+	options: FindManyOptions<WorkflowEntity>,
+): Promise<WorkflowEntity[]> {
+	const workflows = await Db.collections.Workflow.find(options);
```

Comment:
same as for executions we could use `.findAndCount()` here and thus remove the other function